### ISSUE TITLE
Please consider making Changelog more parser friendly (for downstream package managers)

### ISSUE THREAD

**Issue by luzpaz** [ledgersmb/LedgerSMB#7919]
Currently [Changelog](https://raw.githubusercontent.com/ledgersmb/LedgerSMB/master/Changelog) is difficult to parse. A more ideal format would be something like https://keepachangelog.com/

Then it would be easy to ascertain what is the development version. What is the current stable along with release dates for each release. 

Please consider this. TIA!

**Comment by ehuelsmann** [comment]
Hi!

Thanks for raising an issue. I'm aware of https://common-changelog.org/ which goes further than keep-a-changelog in the sense that they indeed specify a format with lots of best practices. Our changelog format is inspired by keep-a-changelog, but doesn't use a Markdown format. Although I understand your remark about the format being hard to parse, it's actually very strictly structured: our release scripts use the formatting requirements to extract the changelog information for the release announcements.

Assuming you're planning to do packaging, were you planning to use specific tools to parse the changelog format? Would it help if we had a utility in our `utils/devel/` directory to extract the information you need, from our changelog? I'm asking because I don't think Markdown is a format that's very well suited for end-user consumption. I mean: it's a great format for writing formatted pages, but this is an example I'm thinking about when referring to MD as unsuitable: https://metacpan.org/dist/Cucumber-TagExpressions/changes and https://metacpan.org/dist/Gherkin/changes. One really needs the HTML rendered version to be able to read it from an end-user perspective (i.e. https://github.com/cucumber/gherkin/blob/main/CHANGELOG.md). One last example: https://github.com/perl-workflow/perl-workflow/blob/master/Changes.md?plain=1 vs https://github.com/perl-workflow/perl-workflow/blob/master/Changes.md.

Instead of having a discussion on formats though, I'll do something better: I'll deliver the parser/extractor which does what you need. Examples:

```plain
ledgersmb$ utils/devel/changelog-parse --series
1.12:1.12.0::unreleased:
1.11:1.11.7:2023-10-03::
1.10:1.10.28:2022-10-08::
1.9:1.9.30:2021-09-24::End-of-Life
1.8:1.8.31:2020-09-04::End-of-Life
1.7:1.7.41:2019-10-04::End-of-Life
1.6:1.6.33:2018-06-10::End-of-Life
1.5:1.5.30:2016-12-24::End-of-Life
1.4:1.4.42:2014-09-15::End-of-Life
1.3:1.3.47:2011-10-12::End-of-Life
1.2:1.2.27:2007-04-04::End-of-Life
1.1:1.1.1:2006-09-20::End-of-Life
1.0:1.0.0p1:2006-09-06::End-of-Life

ledgersmb$ utils/devel/changelog-parse  --series --filter-eol=supported
1.11:1.11.7:2023-10-03::
1.10:1.10.28:2022-10-08::

ledgersmb$ utils/devel/changelog-parse  --release=1.11.7 --changelog
* Fix creating an invoice from an order (#7855)
* Retain partsgroup selection on Update in parts screen (#7848)
* Fix missing columns on trial balance 'Ending' report type (#7870)

```

Hope this is what you were looking for. There are more options to the parser, but please let me know which use-cases you're missing.

**Comment by luzpaz** [comment]
I'm trying to use the automatic release tracking for freshcode.club (http://freshcode.club/submit/ldgersmb)

**Comment by ehuelsmann** [comment]
Thanks for detailing your use-case. From the submission page you link to, I read that you can provide a link to GitHub releases. We post our releases (including "latest" boolean") at https://github.com/ledgersmb/LedgerSMB/releases. That's probably the easiest way to integrate and stay current. Could you check that method of integration?

By the way, your URL misses a letter "e" (ldgersmb instead of ledgersmb). Could you either ask the people at freshcode.club to rename the package or could you create the package at the correct link (too)? Thanks!

**Comment by luzpaz** [comment]
The choice to use github release method option is not super reliable on freshcode, hence my request. Nevertheless, I'll ping the freshcode maintainer and ask if something can be done. 

Thanks for the heads up. the URL is now fixed: http://freshcode.club/projects/ledgersmb

**Comment by luzpaz** [comment]
An aside, a great way to monitor downstream package uniformity is through repology:

<a href="https://repology.org/project/hledger/versions">
    <img src="https://repology.org/badge/vertical-allrepos/hledger.svg?columns=3&header=hledger&exclude_unsupported=1" alt="Packaging status" align="right">
</a>

**Comment by ehuelsmann** [comment]
> The choice to use github release method option is not super reliable on freshcode, hence my request.

If they have an API *and* you don't mind, we could automate posting release updates to them as part of the release announcement, if that works better.

**Comment by ehuelsmann** [comment]
> An aside, a great way to monitor downstream package uniformity is through repology

I'm aware of the packaging situaition. It's terrible. However, the project don't have the tuits to work with Debian and others to get the latest releases in. As a workaround, we've created Docker images to help universal deployment. Meaning to say: we're definitely looking for help on the packaging situation!

**Comment by luzpaz** [comment]
I've ugly regex'd the https://github.com/ledgersmb/LedgerSMB/releases.atom output. This ticket is solved.

**Comment by ehuelsmann** [comment]
For some additional context: there's an installer at https://get.ledgersmb.org now which installs LedgerSMB on a wide range of supported platforms.
